### ISSUE TITLE
Full hard disk due to /tmp/gvmd-split-xml-file-*

### ISSUE THREAD

**Issue by securitytime** [Atomicorp/gvm#48]
I just had the issue that I could not log into the GVM web gui. Having a look at the VM, I noticed that the hard disk was full. Specifically, the /tmp directory took more than 50 GB of space with directories like the following:
```
[root@gvm01 ~]# du -sh /tmp/*
496M    /tmp/gvmd-split-xml-file-0GvOMd
496M    /tmp/gvmd-split-xml-file-0YhUv5
496M    /tmp/gvmd-split-xml-file-118l3u
496M    /tmp/gvmd-split-xml-file-19Hp0A
496M    /tmp/gvmd-split-xml-file-1Uve1k
496M    /tmp/gvmd-split-xml-file-24tRVP
496M    /tmp/gvmd-split-xml-file-2jVS1g
496M    /tmp/gvmd-split-xml-file-2kkKjH
496M    /tmp/gvmd-split-xml-file-2KS2W6
496M    /tmp/gvmd-split-xml-file-2P8Fij
496M    /tmp/gvmd-split-xml-file-2quNF0
[...]
```

I cleared the /tmp directory and rebooted the VM and then GVM was working again. A quick Google search revealed that I am not the only person with this issue. I will post an update if I can find out anything useful.

**Comment by atomicturtle** [comment]
Hmmm, so maybe we need to figure out how to redirect this to another location. Did you catch the timestamp on these? By the location they're clearly temporary (I hope), so we could add a cleanup cron to get rid of these every night. I just have no idea what part of the workflow these are necessary

**Comment by securitytime** [comment]
I did not catch the timestamps. I just checked the /tmp folder again and no such directories reappeared in the last 24 hours. Can you check your /tmp folder? Yes, we could do a cleanup cron. Although I feel this is not an ideal solution. I hope I can reproduce this issue somehow and find out more.

**Comment by securitytime** [comment]
I have another GVM scanner which I installed about a month ago.

I could find another such directory (only one) there.

The timestamp of the directory is from two days ago at 00:17 AM.
However, the timestamp of the split.xml file is from 6 days ago at 06:46 AM.

```
[root@gvm02 ~]# ls -lh /tmp/
[...]
drwx------ 2 gvm  gvm  4.0K Feb 10 00:17 gvmd-split-xml-file-D8p1jm
[...]

[root@gvm02 ~]# ls -lh /tmp/gvmd-split-xml-file-D8p1jm/
total 242M
-rw-r--r-- 1 gvm gvm 242M Feb  6 06:46 split.xml

[root@gvm02 ~]# head /tmp/gvmd-split-xml-file-D8p1jm/split.xml 
<?xml version='1.0' encoding='UTF-8'?>
<cpe-list xmlns="http://cpe.mitre.org/dictionary/2.0" xmlns:ns6="http://scap.nist.gov/schema/scap-core/0.1" xmlns:meta="http://scap.nist.gov/schema/cpe-dictionary-metadata/0.2" xmlns:cpe-23="http://scap.nist.gov/schema/cpe-extension/2.3" xmlns:config="http://scap.nist.gov/schema/configuration/0.1" xmlns:scap-core="http://scap.nist.gov/schema/scap-core/0.3" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://cpe.mitre.org/dictionary/2.0 https://scap.nist.gov/schema/cpe/2.2/cpe-dictionary_2.2.xsd http://scap.nist.gov/schema/cpe-dictionary-metadata/0.2 https://scap.nist.gov/schema/cpe/2.1/cpe-dictionary-metadata_0.2.xsd http://scap.nist.gov/schema/scap-core/0.3 https://scap.nist.gov/schema/nvd/scap-core_0.3.xsd http://scap.nist.gov/schema/configuration/0.1 https://scap.nist.gov/schema/nvd/configuration_0.1.xsd http://scap.nist.gov/schema/scap-core/0.1 https://scap.nist.gov/schema/nvd/scap-core_0.1.xsd">
  <generator>
    <product_name>National Vulnerability Database (NVD)</product_name>
    <product_version>4.2</product_version>
    <schema_version>2.2</schema_version>
    <timestamp>2021-02-06T04:50:45.700Z</timestamp>
  </generator>
  <cpe-item name="cpe:/a:%240.99_kindle_books_project:%240.99_kindle_books:6::~~~android~~">
    <title xml:lang="en-US">$0.99 Kindle Books project $0.99 Kindle Books (aka com.kindle.books.for99) for android 6.0</title
```

Having a look at /var/log/gvm/gvmd.log I found these lines which might be relevant. (My timezone is GMT+1, so 23h17 utc equals the 00:17 AM timestamp from above).
```
md manage:WARNING:2021-02-09 23h19.26 utc:494: insert_nvt_selectors: skipping NVT '1.3.6.1.4.1.25623.1.0.103997' from import of config 'fad93a19-bbcf-4743-978b-eda5eee1eedc' because the NVT does not have a family
md manage:WARNING:2021-02-09 23h19.26 utc:494: create_config_from_file: Internal error
event config:MESSAGE:2021-02-09 23h19.26 utc:494: Scan config could not be created by admin
md manage:WARNING:2021-02-09 23h19.26 utc:494: insert_nvt_selectors: skipping NVT '1.3.6.1.4.1.25623.1.0.103997' from import of config 'cedc9f2f-c86c-428a-bbb4-1df5f0b4ee67' because the NVT does not have a family
md manage:WARNING:2021-02-09 23h19.26 utc:494: create_config_from_file: Internal error
event config:MESSAGE:2021-02-09 23h19.26 utc:494: Scan config could not be created by admin
md manage:WARNING:2021-02-09 23h19.34 utc:162980: split_xml_file: system failed with ret -1, 0 (0), xml_split -s40Mb split.xml && head -n 2 split-00.xml > head.xml && echo '</cpe-list>' > tail.xml && for F in split-*.xml; do    awk 'NR>3 {print last} {last=$0}' $F > body.xml    && cat head.xml body.xml tail.xml > $F;    done
md manage:WARNING:2021-02-09 23h19.34 utc:162980: split_xml_file: and failed to chdir back
md manage:WARNING:2021-02-09 23h19.34 utc:162980: update_scap_cpes: Failed to split CPEs, attempting with full file
md manage:WARNING:2021-02-09 23h19.37 utc:494: insert_nvt_selectors: skipping NVT '1.3.6.1.4.1.25623.1.0.103997' from import of config '89e0fdd6-93e1-481e-94e6-e60d36a005b9' because the NVT does not have a family
md manage:WARNING:2021-02-09 23h19.37 utc:494: create_config_from_file: Internal error
event config:MESSAGE:2021-02-09 23h19.37 utc:494: Scan config could not be created by admin
md manage:WARNING:2021-02-09 23h19.37 utc:494: insert_nvt_selectors: skipping NVT '1.3.6.1.4.1.25623.1.0.103997' from import of config '773b3c3b-9a67-416b-bd45-1aa45ccdc613' because the NVT does not have a family
md manage:WARNING:2021-02-09 23h19.37 utc:494: create_config_from_file: Internal error
event config:MESSAGE:2021-02-09 23h19.37 utc:494: Scan config could not be created by admin
md manage:WARNING:2021-02-09 23h19.48 utc:494: insert_nvt_selectors: skipping NVT '1.3.6.1.4.1.25623.1.0.103997' from import of config 'd777548c-5f85-4d4c-ba34-36a5a5242896' because the NVT does not have a family
md manage:WARNING:2021-02-09 23h19.48 utc:494: create_config_from_file: Internal error
event config:MESSAGE:2021-02-09 23h19.48 utc:494: Scan config could not be created by admin
```

**Comment by katamadone** [comment]
think it's that exact problem:
https://community.greenbone.net/t/skipping-nvt-1-3-6-1-4-1-25623-1-0-150081/8280/3
fixed in 
https://github.com/greenbone/gvmd/pull/1366
which is in the release of 20.8.1
https://github.com/greenbone/gvmd/releases/tag/v20.8.1

**Comment by cfi-gb** [comment]
@katamadone This thread / 20.8.1 is unrelated to the problem currently discussed here (the scan config sync doesn't use the gvmd-split-xml-file*, only the SCAP data sync is doing that).

The solution to install xml_split (on Debian the package is called `xml-twig-tools`) can be found here:

https://community.greenbone.net/t/problems-with-greenbone-feed-sync-gvmd-service/7769/9

This recommendation is (a little bit hidden) documented [here](https://github.com/greenbone/gvmd/blob/v20.8.1/INSTALL.md#prerequisites-for-optional-features):

> Prerequisites (recommended) to lower sync RAM usage
>
> - xml_split (Debian package: xml-twig-tools)

but it seems it is not only lowering sync RAM usage but also improving the sync routine in general.

**Comment by securitytime** [comment]
@cfi-gb 
The xml_split command is already available on my CentOS 8 system under /usr/bin/xml_split.
It belongs to the package perl-XML-Twig.
There is no package named xml_split or xmlsplit available.

**Comment by katamadone** [comment]
Which is not relevant? The NVT Thread from the community looks nearly the same: Skipping NVT, because NVT does not have a family. And because of that the greenbone-feed-sync retries until the disk is full.

**Comment by ghost** [comment]
I am facing the same issue. xml-twig-tools are already installed. I created an issue under the official gvmd repository https://github.com/greenbone/gvmd/issues/1433

**Comment by atomicturtle** [comment]
I'm not seeing these on the atomic-testing 21.04 project, but to be safe I just added a cleanup event at the end of the gvm daily cron to clean any lingering /tmp/gvmd-split-xml-file-* directories

**Comment by atomicturtle** [comment]
closed as resolved in the 21.4.x series
